mmeds/log.py

```diff
@@ -0,0 +1,41 @@
+import logging
+import mmeds.config as fig
+import multiprocessing_logging as mpl
+from multiprocessing import current_process
+
+loggers = {}
+mpl.install_mp_handler()
+
+
+class MMEDSLog():
+    global loggers
+
+    def __init__(self, name, testing=False):
+        if loggers.get(name):
+            self.logger = loggers.get(name)
+        else:
+            self.logger = logging.getLogger('Error')
```

Comment:
logging.getLogger will instantiate the logger with the given name if it doesn't exist already. So you could remove the if ... else logic for instantiating the logger

Reply:
That is what will happen for Multi-Threaded applications, and is what I initially did. But unfortunately it seems the multiprocessing setup doesn't handle this. It was creating a new logger every time a MMEDSLog was instantiated. So I was getting duplicate logs. It's also what was causing an ungodly number of files to be opened during testing. Creating the global dict resolved that issue

Reply:
Ah makes sense. I might add a comment that you had to code around that issue in that case.